haskell-wrapper/tests/ScalarSum.hs

```diff
@@ -0,0 +1,34 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE TypeOperators       #-}
+module ScalarSum (testScalarSum) where
+
+import           Data.Kind                                   (Type)
+import           GHC.TypeNats                                (KnownNat)
+import           Numeric.Natural                             (Natural)
+import           Prelude                                     hiding (Num (..))
+import           RustFunctions                               (rustScalarSum)
+import           Test.Hspec                                  (describe, hspec, it, shouldBe)
+import           Test.QuickCheck                             (Testable (property))
+
+import           ZkFold.Base.Algebra.Basic.Class             (AdditiveSemigroup ((+)))
+import           ZkFold.Base.Algebra.Basic.Field             (Zp)
+import           ZkFold.Base.Algebra.EllipticCurve.BLS12_381 (BLS12_381_G1, BLS12_381_G2)
+import           ZkFold.Base.Algebra.EllipticCurve.Class     (EllipticCurve (ScalarField))
+
+specScalarSum
+    :: forall
+        (a :: Type)
+        (b :: Natural)
+    .   ( ScalarField a ~ Zp b
+        , KnownNat b
+        )
+    =>IO ()
+specScalarSum = hspec $ do
+    describe "Rust add specification" $ do
+        it "should be equal to haskell" $ do
+            property $ \(x :: ScalarField a) (y :: ScalarField a) -> rustScalarSum @a @b x y `shouldBe` (x + y)
+
+testScalarSum :: IO ()
+testScalarSum = do
+    specScalarSum @BLS12_381_G1
+    specScalarSum @BLS12_381_G2
```

Comment:
The scalar field for both G1 and G2 is the same. So, the second test is redundant. Instead, we could test BN254's scalar field here.

Reply:
BN254 is not ready now, @TurtlePU suggested to do it locally for now and then refactor it